docs/modules/ROOT/pages/secrets-management.adoc

```diff
@@ -0,0 +1,154 @@
+
+
+----
+
+TIP: Find more details in http://www.jasypt.org/cli.html[Jasypt CLI Tools documentation]
+
+==== Option 2: using https://www.devglan.com/online-tools/jasypt-online-encryption-decryption[Jasypt Online]
+
+CAUTION: Use this tool at your own risk, since there is a chance of potential leakage of sensitive data
+
+. Paste plain text string to encrypt to the corresponding text field.
+. Set Type of Encryption to "Two Way Encryption (With Secret Text)".
+. Enter secret key/text in the corresponding text field.
+. Click "Encrypt".
+. The resulting encrypted string can be copied from the corresponding field.
+
+=== How to use an encrypted string
+
+. Set the password in one of the following ways:
++
+.. Pass the key with its password-value into a test run using command line:
++
+[source,shell]
+----
+./gradlew runStories -Pvividus.encryptor.password=MYPAS_WORD
+----
++
+.. Add the password to the system project properties as the value of:
++
+[source,properties]
+----
+system.vividus.encryptor.password=MYPAS_WORD
+----
++
+.. Set via OS environment variable `VIVIDUS_ENCRYPTOR_PASSWORD`
++
+Keep in mind that a password specified in the command line takes priority over the project system property,
+and the project system property takes priority over the environment variable.
++
+IMPORTANT: This password must be kept secret and must not be committed to version control system.
++
+NOTE: `MYPAS_WORD` is a sample password and should be never used, own strong password is required for the encryption.
+
+. Use the wrapping `ENC(...)` for any encrypted property value. e.g.
++
+[source,properties]
+----
+http.auth.password=ENC(WVnXhzG704KRZZFYL7hfpA4YoSm8V2F5eNXyr7CRYCichteHvNS3nB23zaI8CuLy)
+----
+
+Please see xref:tests-configuration.adoc#_externalized_configuration[Externalized Configuration section] to get more
+information on how encryptor password can be passed to the tests.
+
+== Secrets Management Tools
+=== https://www.vaultproject.io/[HashiCorp's Vault]
+==== Configuration
+
+Firstly, it is required to configure Vault endpoint and authentication method.
+
+NOTE: The properties marked with *bold* are mandatory.
+
+[cols="1,3,1", options="header"]
+|===
+|Property
+|Description
+|Example
+
+|[subs=+quotes]`*vault.uri*`
+|Vault enpoint
+|`http://127.0.0.1:8200`
+
+|[subs=+quotes]`*vault.token*`
+a|Token authentication requires a static token to be provided.
+[NOTE]
+====
+Token authentication is just one of several authentication methods that are https://docs.spring.io/spring-vault/docs/current/reference/html/index.html#vault.core.environment-vault-configuration[supported].
+====
+|`dev-only-token`
+
+|===
+
+==== How to refer Vault secrets
+. Find the required secrets in Vault.
++
+image::vault.png[Secrets in Vault,width=70%]
+
+. Build full paths to the secrets. For the secrets listed above, the paths would be `secret/vividus/test/username` and
+`secret/vividus/test/password`.
+. Put the built paths to properties using the wrapping `VAULT(...)`
```

Comment:
I think it makes sense to add info that ENC and VAULT are case sensitive even though its upper case in all examples